roles/pul_nomad/tasks/main.yml

```diff
@@ -0,0 +1,190 @@
+---
+# tasks file for pul_nomad
+- name: 'pul_nomad | Install consul'
+  ansible.builtin.include_role:
+    name: 'ansible-consul'
+    public: true
+
+# Set up a runner host
+- name: 'pul_nomad | register a host to run unique commands'
+  set_fact:
+    unique_command_runner: "{{ groups['nomad_servers'][0] }}"
+  run_once: true
+
+# Output unique_command runner
+- name: 'pul_nomad | output unique_command_runner'
+  debug:
+    msg: "unique_command_runner: {{ unique_command_runner }}, inventory_hostname: {{ inventory_hostname }}"
+
+# Set up DNS ACLs
+# Get current tokens
+- name: 'pul_nomad | Register existing tokens'
+  ansible.builtin.shell:
+    cmd: "consul acl token list -format json"
+  environment:
+    CONSUL_HTTP_TOKEN: '{{ consul_acl_master_token }}'
+  register: tokens
+  changed_when: false
+
+- name: 'pul_nomad | Store existing_consul_tokens'
+  set_fact:
+    existing_consul_tokens: "{{ tokens.stdout | from_json | map(attribute='SecretID') }}"
+
+- name: 'nomad-node | Register existing node identities'
+  set_fact:
+    existing_node_names: "{{ tokens.stdout | from_json | community.general.json_query('[*].NodeIdentities[].NodeName') }}"
+
+- name: 'nomad-node | Generate Consul ACL Agent Token'
+  ansible.builtin.shell:
+    cmd: "consul acl token create -node-identity '{{ inventory_hostname_short }}:dc1' -format json"
+  when:
+    - "inventory_hostname_short not in existing_node_names"
+  environment:
+    CONSUL_HTTP_TOKEN: '{{ consul_acl_master_token }}'
+  register: created_agent_token
+  changed_when: false
+
+- name: 'nomad-node | Set Consul ACL Agent Token'
+  ansible.builtin.shell:
+    cmd: "consul acl set-agent-token agent {{ created_agent_token.stdout | from_json | community.general.json_query('SecretID')}}"
+  environment:
+    CONSUL_HTTP_TOKEN: '{{ consul_acl_master_token }}'
+  when:
+    - "inventory_hostname_short not in existing_node_names"
+    - created_agent_token
+
+- name: 'pul_nomad | Create DNS ACL token'
+  ansible.builtin.shell:
+    cmd: "consul acl token create -templated-policy 'builtin/dns' -secret '{{ consul_dns_token }}'"
+  environment:
+    CONSUL_HTTP_TOKEN: '{{ consul_acl_master_token }}'
+  when:
+    - "unique_command_runner == inventory_hostname"
+    - "consul_dns_token not in existing_consul_tokens"
+
+- name: 'pul_nomad | Assign DNS ACL token'
+  ansible.builtin.shell:
+    cmd: "consul acl set-agent-token dns '{{ consul_dns_token }}'"
+  environment:
+    CONSUL_HTTP_TOKEN: '{{ consul_acl_master_token }}'
+
+# Consul setup for nomad
+
+# tasks file for pulibrary-nomad-node
+- name: Add Docker GPG apt Key
```

Comment:
When we sent this PR we had ditched podman because the podman folks will not update their package in the ubuntu repository, and that's a security nightmare - so we abandoned it. Then Francis was like, we could have up-to-date Podman if we used Rocky, and that work is here: https://github.com/pulibrary/princeton_ansible/pull/5098

Reply:
we needed you to sign off on this @acozine ;-)

Reply:
Will be implemented by #5098